### PR TITLE
federation api version v1alpha1 to v1beta1

### LINKED ISSUE #856: Update version of APIs that are transitioning to beta
Any API that we intend to move to a beta status will need to have its version updated in the code and documentation before we release.

### PR DESCRIPTION
**Pull request by xunpan**
Fixes #856

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -63,12 +63,12 @@ accomplished as follows:
 
 ```bash
 # Bootstrap and commit a new type
-$ kubebuilder create api --group <your-group> --version v1alpha1 --kind <your-kind>
+$ kubebuilder create api --group <your-group> --version v1beta1 --kind <your-kind>
```
Comment (irfanurrehman) [inline docs/development.md]: I don't think this should change. Bootstrapping a new API group should still land into alpha.

**Comment by xunpan** [comment]
@irfanurrehman 
Based on my understanding, all API is moved to Beta for our release. If it is not right, please let me know what is the standard to make an API to beta? I may lost related information in my mind.

To make an API to beta, it is string replacement to me. What kind of things that is not string replacement in such change? This is first time I move API from alpha to beta. Am I lost anything?

**Comment by xunpan** [comment]
Reference material of kubernetes: https://kubernetes.io/docs/concepts/overview/kubernetes-api/

**Comment by irfanurrehman** [comment]
> @irfanurrehman
> Based on my understanding, all API is moved to Beta for our release. If it is not right, please let me know what is the standard to make an API to beta? I may lost related information in my mind.
> 
> To make an API to beta, it seems string replacement to me. What kind of things that is not string replacement in such change? This is first time I move API from alpha to beta. Am I lost anything?

Please update the changes as discussed offline, post which we can exchange more thoughts. :-)

**Comment by marun** [comment]
@xunpan This LGTM, but a rebase is required before I can merge.

**Comment by xunpan** [comment]
@marun rebase done
And resource status kept in alpha

File: pkg/apis/core/v1beta1/federatedtypeconfig_types.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
```
Comment (font) [inline pkg/apis/core/v1beta1/federatedtypeconfig_types.go]: We should probably add a GoDoc comment suggesting the `StatusType` and `StatusCollection` fields are still in alpha.

Comment (marun) [reply pkg/apis/core/v1beta1/federatedtypeconfig_types.go]: I don't think this needs to be handled in this PR, and I've filed #918 to track it and related work.

**Comment by marun** [comment]
Ack @xunpan Please rebase.

**Comment by xunpan** [comment]
@marun rebase done

**Comment by marun** [comment]
Thank you! 

/lgtm

**Pull request merged**